modules/src/clients/ics13_near/types.rs

```diff
@@ -47,16 +47,14 @@ impl Signature {
         }
     }
 
-    // TODO: we might want to create a trait for signature verification
-    // or integrate this into HostFunctions
     pub fn verify<T: HostFunctionsProvider>(
         &self,
-        data: impl AsRef<[u8; 32]>,
-        public_key: PublicKey,
+        data: &[u8; 32],
+        public_key: &PublicKey,
     ) -> bool {
         match self {
-            Self::Ed25519(signature) => T::ed25519_recover(signature.as_ref(), data.as_ref())
-                .map(|key| key == public_key.0.as_ref())
+            Self::Ed25519(signature) => T::ed25519_recover(signature.as_ref(), data)
```

Comment:
Not really sure about recovering ed25519 public keys from signatures.
Check this article for more context https://wiki.polkadot.network/docs/learn-keys.

That's why public key recovery for Schnorr signatures are not even implemented on substrate.

Reply:
Also mentioned in the last section here https://cryptobook.nakov.com/digital-signatures/eddsa-and-ed25519#:~:text=The%20produced%20digital%20signature%20is,%2B%2057%20bytes)%20for%20Ed448.

Reply:
I've done this in the PoC: https://github.com/ComposableFi/near-trustless-bridge/blob/b471d7fe7dc3bd692036b10b53ab465d12fa0750/near-lite-client/src/signature.rs#L9
Not sure why @seunlanlege chose to do ti through `ed25519_recover`.

Reply:
fixed in [2ee0220](https://github.com/ComposableFi/ibc-rs/pull/8/commits/2ee0220f5940fae191d17b8d198be4b67a7e6ee8)